modules/simple_forms_api/docs/rfcs/0003-form-submission-purge-plan.md

```diff
@@ -0,0 +1,115 @@
+# RFC: Data Purge Plan for `FormSubmission` Records
+
+## 1. Title
+
+Data Purge Plan for `FormSubmission` Records
+
+## 2. Summary
+
+This document outlines a data purge plan for `FormSubmission` records stored in the VA.gov PostgreSQL database. The goal is to ensure compliance with best practices for handling Personally Identifiable Information (PII) and Protected Health Information (PHI) by removing sensitive data after a specified retention period while maintaining essential metadata for operational use. The purge process will also include deleting associated PDF backups stored in an AWS S3 bucket.
+
+## 3. Background
+
+The `FormSubmission` table contains user-submitted data, including PII and PHI. Associated PDF backups are also stored in a team's S3 bucket. Current retention policies suggest removing user and form data after 60 days post-submission. While the form data is no longer required for operations, minimal metadata (`submission_date` and `benefits_intake_uuid`) will be retained for reporting and referencing. This purge process will help mitigate risks related to prolonged retention of sensitive data while maintaining functionality for users to download their PDF backups during the retention period.
+
+## 4. Proposal
+
+### 4.1. Data Retention and Purge Process
+
+- **Retention Policy**:
+
+  - Purge sensitive data from `FormSubmission` records 60 days after the `updated_at` once `FormSubmission.submission_status` has been set to `vbms`.
+  - Retain:
+    - `submission_date` for operational timelines.
+    - `benefits_intake_uuid` for linking associated PDFs as well as historical lookups via the Benefits Intake API.
+  - Leave the `FormSubmission` record intact for future reference.
+
+- **PDF Purge**:
+
+  - Use `benefits_intake_uuid` to identify and delete PDFs from each team's AWS S3 bucket.
+
+- **Trigger for Purge**:
+  - Purge records where the status is `"vbms"`, marking the final stage of submission processing.
+  - For records not reaching `"vbms"`, resolve via manual remediation or notify users.
+
+### 4.2. Job Implementation
+
+- **Scheduled Purge Job**:
+
+  - Utilize Sidekiq to run a background job daily during off-peak hours.
+  - Job steps:
+    1. Query `FormSubmission` records with an `updated_at` older than 60 days and a status of `"vbms"`.
+    2. Purge sensitive data from identified records.
+    3. Delete associated PDFs from S3.
+
+- **Delay Between Marking and Deleting**:
```

Comment:
I would vote for just delete for simplicity. Unless there's a good reason to mark it first.

Reply:
I could see another use case for delaying and that would be for jobs to "listen" for this and do something upon deletion.

I'm not familiar with the specifics so this may not be a use case but as I read this I wondered if an `ActiveSupport::Notification` should be emitted when deleting PII (which wouldn't require a delay).

Example:
```ruby
ActiveSupport::Notifications.instrument(
  "pii.deleted", 
  {
    record_type: "FormSubmission",
    benefits_intake_uuid: form_submission.benefits_intake_uuid,
    deleted_at: Time.current
  }
)
```

These could be used for audit trails, metrics, or additional clean up jobs.

Just a suggestion; I'm mostly just throwing an idea out there.

Reply:
I love this idea, thanks @ericboehs!